File: ui/src/main/java/ru/vertuos/ui/animation/container/AnimationContainer.java
package ru.vertuos.ui.animation.container;

import com.badlogic.gdx.graphics.g2d.Animation;

import java.util.HashMap;
import java.util.Map;

public class AnimationContainer<T> {

    private final Map<String, Animation<T>> animationMap;

    public AnimationContainer() {
        this.animationMap = new HashMap<>();
    }

    public void putAnimation(String key, Animation<T> animation) {
        animationMap.put(key, animation);
    }

    public Animation<T> remove(String key) {
        return animationMap.remove(key);
    }

    public Animation<T> getAnimation(String key) {
        return animationMap.get(key);
    }
}

File: ui/src/main/kotlin/ru/vertuos/ui/animation/container/AnimationContainer.kt
package ru.vertuos.ui.animation.container

import com.badlogic.gdx.graphics.g2d.Animation

class AnimationContainer<T> {

    private val animationMap: MutableMap<String, Animation<T>> = HashMap()

    fun putAnimation(key: String, animation: Animation<T>) {
        animationMap[key] = animation
    }

    fun remove(key: String): Animation<T>? {
        return animationMap.remove(key)
    }

    fun getAnimation(key: String): Animation<T>? {
        return animationMap[key]
    }
}
